packages/shylu/shylu_dd/frosch/src/CoarseSpaces/FROSch_CoarseSpace_def.hpp

```diff
@@ -138,16 +139,21 @@ namespace FROSch {
                             FROSCH_ASSERT(NumVectors_i+itmp <= AssembledBasis_->getNumVectors(),"FROSch::CoarseSpace: NumVectors_i+itmp <= AssembledBasis_->getNumVectors()");
                             FROSCH_ASSERT(LocalLength_i+Offsets_[i] <= AssembledBasis_->getLocalLength(),"FROSch::CoarseSpace: LocalLength_i+Offsets_[i] <= AssembledBasis_");
 
+                            Kokkos::RangePolicy<execution_space> policy (0, LocalLength_i);
+                            // Xpetra wrapper for Tpetra MV
+                            auto unassembledXTpetraMVector = rcp_dynamic_cast<const TpetraMultiVector<SC,LO,GO,NO>>(UnassembledSubspaceBases_[i]);
+                            auto   assembledXTpetraMVector = rcp_dynamic_cast<      TpetraMultiVector<SC,LO,GO,NO>>(AssembledBasis_);
+                            // Tpetra MV
+                            auto unassembledTpetraMVector = unassembledXTpetraMVector->getTpetra_MultiVector();
+                            auto   assembledTpetraMVector = assembledXTpetraMVector->getTpetra_MultiVector();
+                            // Views
+                            auto unassembledView = unassembledTpetraMVector->getLocalViewDevice(Tpetra::Access::ReadOnly);
+                            auto   assembledView =   assembledTpetraMVector->getLocalViewDevice(Tpetra::Access::ReadWrite);
```

Comment:
`ReadWrite` or `OverwriteAll`?

Reply:
I think it does need to be ReadWrite because only rows Offset_i ... Offset_i + LocalLength_i are updated in each MV.